File: 03460.py
for i in range(int(input())):
    bin = [0 for i in range(21)]
    n = int(input())

    two = 2; index = 1
    while two <= n:
        two *= 2; index += 1
    two /= 2; index -= 1

    while n > 0:
        if two <= n:
            n -= two
            bin[index] = 1

        two /= 2
        index -= 1

    for j in range(21):
        if bin[j]:
            print(j, end=' ')
    print()

File: 03460.cpp
#include <iostream>
using namespace std;

int main() {
	int T; cin >> T;

	for (int i = 0; i < T; i++) {
		int bin[21] = { 0, };
		int n; cin >> n;

		int two = 2, index = 1;
		while (two <= n) { two *= 2; index++; }
		two /= 2; index--;

		while (n > 0) {
			if (two <= n) {
				n -= two;
				bin[index] = 1;
			}

			two /= 2;
			index--;
		}

		for (int i = 0; i < 21; i++)
			if (bin[i]) cout << i << ' ';
		cout << endl;
	}
}
